include/upipe/ubuf_block.h

```diff
@@ -479,6 +479,50 @@ static inline int ubuf_block_resize(struct ubuf *ubuf, int offset, int new_size)
     return UBASE_ERR_NONE;
 }
 
+/** @This extends a block ubuf, if possible. This will only work if
+ * the relevant low-level buffers are not shared with other ubuf and
+ * the block manager allows to grow the buffer (ie. prepend/append have been
+ * correctly specified at allocation, or reallocation is allowed)
+ *
+ * Should this fail, @ref ubuf_block_merge may be used to achieve the same
+ * goal with an extra buffer copy.
+ *
+ * @param ubuf pointer to ubuf
+ * @param prepend number of octets to prepend
+ * @param append number of octets to append
+ * @return an error code
+ */
+static inline int ubuf_block_extend(struct ubuf *ubuf, int prepend, int append)
+{
+    assert(prepend >= 0);
+    assert(append >= 0);
+    if (ubuf->mgr->signature != UBUF_ALLOC_BLOCK)
+        return UBASE_ERR_INVALID;
+
+    if (!ubase_check(ubuf_control(ubuf, UBUF_SINGLE)))
+        return UBASE_ERR_BUSY;
+
+    struct ubuf_block *block = ubuf_block_from_ubuf(ubuf);
+    if (prepend > block->offset)
+        return UBASE_ERR_INVALID;
+    if (append) {
```

Comment:
I am not sure we need to get into that logic. Just see if there is enough contiguous space to expend the block, or bail out and let the caller call ubuf_block_merge. This would also get rid of the additional routine in ubuf_block_mem.c.